play-json-tools/src/test/scala/com/evolution/playjson/tools/FlatFormatSpec.scala

```diff
@@ -30,14 +30,14 @@ class FlatFormatSpec extends AnyFunSuite with Matchers {
       FlatFormat("inner2", Json.format[Outer])
     }
   }
+}
 
-  case class Inner(inner: String)
```

Comment:
This is weird: when these classes are inside `class FlatFormatSpec`, the Scala 3 compiler crashes when expanding the macro. I found a similar issue: https://github.com/sirthias/borer/issues/585. Their solution (move classes out of enclosing class) helped 🤷‍♂️

Reply:
That's because of reference to outer class (trait in their case).

Reply:
Yes, I'd expect it either to work (since the class is the same, only its location differs) or at least produce a comprehensible error, to be honest, but neither is the case here, unfortunately 🤷